integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/ui/dialogs/CoinBaseResultDialog.kt

```diff
@@ -73,11 +74,12 @@ class CoinBaseBuyDashDialog : DialogFragment() {
             }
 
             binding.coinbaseBuyDialogPositiveButton.setOnClickListener {
-                onCoinBaseBuyDashDialogButtonsClickListener?.onPositiveButtonClick(Type.values().first { it.ordinal == type })
+                onCoinBaseResultDialogButtonsClickListener?.onPositiveButtonClick(Type.values().first { it.ordinal == type })
             }
         }
 
         binding.coinbaseBuyDialogNegativeButton.setOnClickListener {
+            type?.let { onCoinBaseResultDialogButtonsClickListener?.onNegativeButtonClick(Type.values().first { it.ordinal == type }) }
             dismiss()
             findNavController().popBackStack()
             findNavController().popBackStack()
```

Comment:
Some navigation is handled here, and some by the callers. This makes implementing events difficult.
There is also a sheet of methods that resolve what text will be shown in this dialog depending on where it's called from. The more callers there are, the longer the sheet.

This seems like a bad design. The caller should resolve the text and handle the actions, and this dialog should only serve as a generic UI container. This will also allow to reuse it in other parts of the app like CrowdNode and other integrations.